tools/extract_ir/extract_ir.py

```diff
@@ -53,37 +56,39 @@ def parse_args():
                         required=False,
                         default='',
                         help='The file to output to, defaults to stdout.')
-    parser.add_argument('--llvm-dis',
+    parser.add_argument('-d', '--disassembler',
                         required=False,
                         default='',
-                        help='The path to llvm-dis. Bytecode will be disassembled only if this is provided.')
+                        help='''The path to llvm-dis or spirv-dis. Bytecode will
+                                be disassembled only if this is provided.''')
     return parser.parse_args()
 
 
-def try_disassemble_spir(llvm_dis_path, input_file_name, output_file_name='-'):
+def disassemble_spir(disassembler_path, input_file_name, output_file_name):
     """
-    Given to file names, tries to use llvm-dis to disassemble the input file into the output file.
-    The output file defaults to stdout.
+    Given two file names, uses a disassembler to disassemble the input file into
+    the output file.
     """
     code = subprocess.call(
-        [llvm_dis_path,
-         input_file_name,
-         '-o=' + output_file_name])
+        [disassembler_path, '-o ' + output_file_name, input_file_name])
```

Comment:
If you pass "-o" and the filename as one string like this, you end up with both in one element of argv (e.g. argv[2] == "-o -" which leads to "unknown command line option" errors from llvm-dis. If you instead change it to "-o", "-", it'll work.